### PR TITLE
add support for teams 

### PR DESCRIPTION
**Pull request by martynjarvis**
The previous library only supported 1 person-teams free-for-alls. This meant that in 2 vs 2 games the losing team were treated as if they had drawn with each other which could cause low ranked losing players to actually increase in rank.

This PR implements a new trueskill library supports N vs N games (e.g. 1 vs 1 and 2 vs 2). It can also support uneven teams (e.g. 1 vs 2).

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -89,29 +89,35 @@ function getAll(){
...
+            console.log("Unsupported game winner ", game.winner);
+        }
+
+        var teamL = ps.filter(p=>game.playersL.includes(p.id));
+        var teamR = ps.filter(p=>game.playersR.includes(p.id));
+        var otherPlayers = ps.filter(
+            p=>!game.playersL.includes(p.id) && !game.playersR.includes(p.id));
+
+        var newRatings = calculateNewRatings(
+            GameInfo, teamL, teamR, rankL, rankR);
```
Comment (leontrolski) [inline main.js]: I think `trueskill.js` should have at least a fake module style export, as in it should just end with `trueSkill2 = {calculateNewRatings, GameInfo}` and then here do `trueSkill2.calculateNewRatings(...)`

File: main.js [hunk 2]
```diff
@@ -89,29 +89,35 @@ function getAll(){
...
-            .filter(p=>losers.includes(p.id))
-            .map(p=>({rank: 2, id:p.id, skill:p.skill}))
-        var toAdjust = [...winners, ...losers]
-        trueskill.AdjustPlayers(toAdjust)
-        return ps.map(p=>({
-            id: p.id,
-            skill: allParticipants.includes(p.id)?
-                toAdjust.filter(o=>o.id===p.id)[0].skill : p.skill
-        }))
+        var rankL, rankR;
```
Comment (leontrolski) [inline main.js]: in a spirit of error message free minimalism, could replace all this with `var [rankL, rankR] = game.winner === 'l'? [1, 2] : [2, 1]`

File: main.js (hunk 1 above)
Comment (leontrolski) [reply main.js]: Also, given we will never have the concept of a draw, would it be better for `calculateNewRatings` to just take `winners` and `losers`. (`GameInfo` could also remain private to `trueSkill2`).

File: main.js [hunk 3]
```diff
@@ -89,29 +89,35 @@ function getAll(){
...
+            p=>!game.playersL.includes(p.id) && !game.playersR.includes(p.id));
+
+        var newRatings = calculateNewRatings(
+            GameInfo, teamL, teamR, rankL, rankR);
+
+        return [...newRatings, ...otherPlayers]
     }
     // initial player values
-    var ps = players.map(player=>({id: player, skill: [25.0, 25.0/3.0]}))
+    var ps = players.map(player=>({id: player, mean: 25.0, stdev: 25.0/3.0}))
```
Comment (leontrolski) [inline main.js:103]: (thumbsup)

File: trueskill.js [hunk 4]
```diff
@@ -1,886 +1,131 @@
...
+function update(gameInfo, newRatings, thisTeam, otherTeam, result){
+    let drawMargin = 0; // We can't really draw a fussball game can we?
+    let betaSquared = gameInfo.beta * gameInfo.beta;
+    let tauSquared = gameInfo.tau * gameInfo.tau;
+    let totalPlayers = thisTeam.length + otherTeam.length;
 
-  Wdraw = function(t, e) {
-    return pow(Vdraw(t, e), 2) + ((e - t) * pdf(e - t) + (e + t) * pdf(e + t)) / (cdf(e - t) - cdf(-e - t));
-  };
+    let thisMeanSum = thisTeam.reduce((total, x) => total + x.mean, 0);
```
Comment (leontrolski) [inline trueskill.js]: just define `sum` brah

File: main.js (hunk 1 above)
Comment (martynjarvis) [reply main.js]: Maybe, but that just pushes calculating winners and losers to `main.js`. It makes sense to keep it here, especially if we want to extend to include the concept of a draw.

Comment (leontrolski) [reply main.js]: Martyn, we're never gonna have a draw bro

**Pull request merged**